### PR TITLE
Add CGAL::CGAL_Basic_viewer target

### PR DESCRIPTION
**Pull request by sloriot**
Simplify the creation of program using the basic viewer

### CODE REVIEW

**Comment by sloriot** [comment]
There is a problem when the target is used in `Surface_mesh_topology` tests with `-DWITH_tests=ON` but I don't understand why as I don't see any difference with examples (that are fine). Maybe @lrineau has a clue.

File: Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/draw_polygon_set_2.h [hunk 1]
```diff
@@ -3,7 +3,8 @@ namespace CGAL {
 /*!
 \ingroup PkgDrawPolygonSet2
 
-opens a new window and draws `aps`, an instance of the `CGAL::Polygon_set_2` class. A call to this function is blocking, that is the program continues as soon as the user closes the window. This function requires CGAL_Qt5, and is only available if the flag CGAL_USE_BASIC_VIEWER is defined at compile time.
+opens a new window and draws `aps`, an instance of the `CGAL::Polygon_set_2` class. A call to this function is blocking, that is the program continues as soon as the user closes the window. This function requires `CGAL_Qt5`, and is only available if the macro `CGAL_USE_BASIC_VIEWER` is defined.
+Linking with the cmake target `CGAL::CGAL_Basic_viewer` will link with `CGAL_Qt5` add the definition.
```
Comment (gdamiand) [inline Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/draw_polygon_set_2.h:7]: Suggestion for the last sentence:

Linking with the cmake target `CGAL::CGAL_Basic_viewer` will link with `CGAL_Qt5` and add the `CGAL_USE_BASIC_VIEWER` definition.

File: Straight_skeleton_2/examples/Straight_skeleton_2/CMakeLists.txt [hunk 2]
```diff
@@ -9,10 +9,6 @@ find_package(CGAL REQUIRED COMPONENTS Core)
 
 find_package(CGAL COMPONENTS Qt5)
```
Comment (gdamiand) [inline Straight_skeleton_2/examples/Straight_skeleton_2/CMakeLists.txt:10]: find_package(CGAL REQUIRED OPTIONAL_COMPONENTS Qt5) ?

File: Straight_skeleton_2/examples/Straight_skeleton_2/CMakeLists.txt [hunk 3]
```diff
@@ -9,10 +9,6 @@ find_package(CGAL REQUIRED COMPONENTS Core)
...
 
-if(CGAL_Qt5_FOUND)
-  add_definitions(-DCGAL_USE_BASIC_VIEWER -DQT_NO_KEYWORDS)
-endif()
-
 include(${CGAL_USE_FILE})
```
Comment (gdamiand) [inline Straight_skeleton_2/examples/Straight_skeleton_2/CMakeLists.txt:12]: Remove include(${CGAL_USE_FILE}) 
(I though we already removed all these use file ?)

**Review by gdamiand — APPROVED** [review]

**Comment by gdamiand** [comment]
> There is a problem when the target is used in `Surface_mesh_topology` tests with `-DWITH_tests=ON` but I don't understand why as I don't see any difference with examples (that are fine). Maybe @lrineau has a clue.

I tested, and observed the same problem. There are probably some differences between examples and tests for our cmake?

File: Straight_skeleton_2/examples/Straight_skeleton_2/CMakeLists.txt (hunk 3 above)
Comment (lrineau) [reply Straight_skeleton_2/examples/Straight_skeleton_2/CMakeLists.txt:12]: > Remove include(${CGAL_USE_FILE}) (I though we already removed all these use file ?)

That line is useless, but we have not yet made a pass over all `CMakeLIists.txt` to modernize them.

**Comment by lrineau** [comment]
> > There is a problem when the target is used in `Surface_mesh_topology` tests with `-DWITH_tests=ON` but I don't understand why as I don't see any difference with examples (that are fine). Maybe @lrineau has a clue.
> 
> I tested, and observed the same problem. There are probably some differences between examples and tests for our cmake?

Me too: I reproduce, but I do not understand why there is a difference in behavior.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.4-Ic-87

**Review by lrineau — APPROVED** [review]

**Pull request merged**